### PR TITLE
Fix libCling compilation on Windows

### PR DESCRIPTION
**Pull request by bellenot**
- add dependencied on Core and IO (needed to resolve the symbols at link time on Windows)
- add a few symbols to be exported
- use the ANSI version of system functions
- convert the backslashs to forward slashs in the rootmap file path

### CODE REVIEW

File: core/metacling/src/CMakeLists.txt [hunk 1]
```diff
@@ -38,8 +38,23 @@ ROOT_LINKER_LIBRARY(Cling
         $<TARGET_OBJECTS:MetaCling>
         LIBRARIES ${CLING_LIBRARIES})
 
+if(MSVC)
+  set_target_properties(Cling PROPERTIES WINDOWS_EXPORT_ALL_SYMBOLS TRUE)
+  set(cling_exports ??1Value@cling@@QAE@XZ
+      ?setValueNoAlloc@internal@runtime@cling@@YAXPAX00D_K@Z
+      ?printValue@cling@@YA?AV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@PB_N@Z
+      ?EvaluateDynamicExpression@internal@runtime@cling@@YA?AVValue@3@PAVInterpreter@3@PAVDynamicExprInfo@123@PAVDeclContext@clang@@@Z
+  )
```
Comment (amadio) [inline core/metacling/src/CMakeLists.txt]: This looks quite hacky, is it the only way?

Comment (bellenot) [reply core/metacling/src/CMakeLists.txt]: Well, there are always several solutions for a given problem, but I would say this is the simplest one

File: core/metacling/src/TCling.cxx [hunk 2]
```diff
@@ -4729,20 +4728,24 @@ int TCling::ReadRootmapFile(const char *rootmapfile, TUniqueString *uniqueString
    const std::map<char, unsigned int> keyLenMap = {{'c',6},{'n',10},{'t',8},{'h',7},{'e',5},{'v',4}};
 
    if (rootmapfile && *rootmapfile) {
-
+      std::string rmapfile(rootmapfile);
```
Comment (Axel-Naumann) [inline core/metacling/src/TCling.cxx]: Can you call that `rootmapfileNoBackslash`?

**Comment by bellenot** [comment]
bloody f*****g git :-1:

**Pull request merged**